mod_reforged/hooks/skills/effects/adrenaline_effect.nut

```diff
@@ -0,0 +1,37 @@
+::Reforged.HooksMod.hook("scripts/skills/effects/adrenaline_effect", function(q) {
+	q.m.GrantsInjuryImmunity <- true;
+
+	q.create = @(__original) function()
+	{
+		__original();
+
+		this.m.Order = ::Const.SkillOrder.PermanentInjury - 100;	// This effect needs to flip the IsAffectedByInjuries flag before any of the injuries had the time to inflict their effects
+	}
+
+	q.getTooltip = @(__original) function()
+	{
+		local ret = __original();
+
+		if (this.m.GrantsInjuryImmunity)
+		{
+			ret.push({
+				id = 11,
+				type = "text",
+				icon = "ui/icons/special.png",
+				text = "Ignore the effects of non-permanent injuries"
+			});
+		}
+
+		return ret;
+	}
+
+	q.onUpdate = @(__original) function( _properties )
+	{
+		__original(_properties);
+
+		if (this.m.TurnsLeft != 0 && this.m.GrantsInjuryImmunity)
+		{
+			_properties.IsAffectedByInjuries = false;
```

Comment:
This will also make it impossible for the character to receive injuries (actor.nut `onDamageReceived` function checks this flag in the target's `CurrentProperties` and if false then it doesn't apply an injury). Adding a check for whether this entity is the currently active entity is also not enough because that will mean he can't receive injury from enemy's riposte.

So you need to do it differently i.e. reset this flag to its original value before taking damage. For ideas consider looking at my implementation of the `rf_undead_injury_receiver_effect` which does the opposite i.e. makes undead be affected by injuries but only receive them if attacker has crippling strikes perk.

Reply:
Over at MSU I've been trying to figure out a clean solution for modularizing the receiving of injuries from injuries applying their effects which also is compatible with vanilla and so on. But there are certain edge cases so it will need some more time/effort to find the best solution.

For now, I suggest we make this a feature of Indomitable i.e. under the effects of indomitable you cannot receive and are not affected by already present temporary injuries (so basically the same as the vanilla iron will potion).

Reply:
In Discord we approved this "feature". So you can update the strings and tooltips in this PR to say that it also makes you immune to receiving injuries.